File: haskell/0006.hs
sumOfSquares :: (Integral a) => a -> a
sumOfSquares limit = sum [x^2 | x <- [1..limit]]

squareOfSums :: (Integral a) => a -> a
squareOfSums limit = (sum [1..limit]) ^ 2

main :: IO ()
main = putStrLn $ show $ (squareOfSums 100) - (sumOfSquares 100)

File: elixir/lib/0006.ex
defmodule ProjectEuler.P0006 do
  def solve(limit) do
    square_of_sum = 1..limit
      |> Enum.to_list
      |> Enum.sum
    square_of_sum =  square_of_sum * square_of_sum

    sum_of_squares = 1..limit
      |> Enum.to_list
      |> Enum.map(&(&1 * &1))
      |> Enum.sum

    square_of_sum - sum_of_squares
  end

  def print do
    solve 100
  end
end
